source/expressions.tex

```diff
@@ -6771,21 +6771,18 @@
 right operand is a bit-field.
 
 \pnum
-In contexts where comma is given a special meaning,
-\begin{example}
-in
-lists of arguments to functions\iref{expr.call} and lists of
-initializers\iref{dcl.init}
-\end{example}
-the comma operator as
-described in this subclause can appear only in parentheses.
+\begin{note}
+In contexts such as function calls\iref{expr.call}, lists of initializers\iref{dcl.init},
+or \grammarterm{template-argument-list}{s}\iref{temp.names},
+the comma operator as described in this subclause can appear only in parentheses.
```

Comment:
The old wording only applied to places where comma has special meaning; the new (non-normative, but still) wording appears to also apply even in subexpressions where the special meaning is removed. For example:

```
int a = true ? 1, 2 : 3;  // disallowed by this note, because comma is not in parentheses
int arr[3];
int b = arr[1, 2];  // ok but deprecated, disallowed by this note
int c = []{ foo(), bar(); return 0; }(); // comma operator disallowed by this note
```

Can we preserve the "where comma is given a special meaning" part, to avoid giving a false impression of the rules here?